### PR TITLE
[ISSUE-15] Add offline e2e fixtures and modernize core CI (pytest + Ruff)

### PR DESCRIPTION
**Pull request by Gonza10V**
### Motivation
- Provide v1 end-to-end, offline, compiler-only pytest fixtures that exercise the full-build orchestration and keep automation/hardware tests optional and manual.
- Modernize core CI to a fast, local, deterministic pipeline based on `pytest` and `ruff` that runs only compiler tests (`unit`, `stages`, `integration`).
- Avoid requiring network/SynBioHub/PUDU/Opentrons for core CI and preserve the new clean-architecture package layout for testing.

### Description
- Add a GitHub Actions workflow at `.github/workflows/ci.yml` that installs the package with test extras and runs `ruff check .`, `ruff format --check .`, and `pytest tests/unit tests/stages tests/integration` on Python 3.10/3.11.
- Modernize the test extra in `pyproject.toml` to `pytest>=7,<9` and add a pytest marker configuration for `automation` tests under `[tool.pytest.ini_options]`.
- Add shared fixtures in `tests/conftest.py` (in-memory `sbol2.Document`, `default_build_options`, `minimal_lvl2_request`) and add integration tests `tests/integration/test_full_build_happy_path.py` and `tests/integration/test_missing_lvl1_then_domestication.py` that run offline using fake stage implementations.
- Add stage smoke tests under `tests/stages/`, a unit import boundary test `tests/unit/test_core_imports.py` (ensures optional automation deps are not side-effect imported), optional/manual automation placeholder `tests/automation/test_opentrons_simulation_optional.py`, and docs `tests/README.md` and `RELEASE_CHECKLIST.md`.

### Testing
- Ran core import checks with `python -c "from buildcompiler.api import BuildCompiler, full_build"`, `python -c "from buildcompiler.execution import BuildContext, FullBuildExecutor"`, and `python -c "from buildcompiler.reporting import BuildSummary, BuildReport, BuildGraph"` and the PUDU adapter imports; these succeeded.
- Ran `pytest tests/unit tests/stages tests/integration` and observed all core tests pass (`89 passed` in ~0.44s) verifying the new integration and orchestration fixtures succeeded.
- Ran `ruff check .` as part of validation and it failed due to pre-existing repository lint debt outside this issue scope (legacy files/notebooks); this is noted and left for separate cleanup.
- Attempted `uv run ...` verification and it failed while installing the optional automation git dependency `SBOLInventory` due to environment/network restrictions (git fetch HTTP 403), confirming automation extras remain optional and are not required by core CI.

------
[Codex Task](https://chatgpt.com/codex/cloud/tasks/task_b_69fb94193230832396a70cc5fa03a650)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+          python-version: ${{ matrix.python-version }}
+      - name: Install package
+        run: |
+          python -m pip install -U pip
+          python -m pip install -e ".[test]"
+          python -m pip install ruff
+      - name: Ruff check
+        run: ruff check .
+      - name: Ruff format check
+        run: ruff format --check .
```
[the remark this answers is not in the document]
Comment (Gonza10V) [reply .github/workflows/ci.yml]: @codex good catch please fix it
